slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/validator/GenesisValidatorRootValidator.java

```diff
@@ -57,16 +70,15 @@ public boolean genesisValidatorRootExists() {
                 handle -> metadataDao.findGenesisValidatorsRoot(handle).isPresent()));
   }
 
-  private boolean validateGvr(final Handle handle, final Bytes32 genesisValidatorsRoot) {
+  private Void findAndInsertGVR(final Handle handle, final Bytes32 genesisValidatorsRoot) {
```

Comment:
Think this is 'Or' not 'And'.

```suggestion
  private Void findOrInsertGVR(final Handle handle, final Bytes32 genesisValidatorsRoot) {
```

Reply:
renamed it to `findAndInsertIfNotExists` which is more clear in the intent.